website/common/script/ops/buy/abstractBuyOperation.js

```diff
@@ -0,0 +1,133 @@
+import i18n from '../../i18n';
+import {
+  NotAuthorized,
+} from '../../libs/errors';
+import _merge from 'lodash/merge';
+import _get from 'lodash/get';
+
+export class NotImplementedError extends Error {
+  constructor (str) {
+    super(`Method: '${str}' not implemented`);
+  }
+}
+
+export class AbstractBuyOperation {
+  /**
+   * @param {User} user - the User-Object
+   * @param {Request} req - the Request-Object
+   * @param {analytics} analytics
+   */
+  constructor (user, req, analytics) {
+    this.user = user;
+    this.req = req || {};
+    this.analytics = analytics;
+
+    this.quantity = _get(req, 'quantity', 1);
+  }
+
+  /**
+   * Shortcut to get the translated string without passing `req.language`
+   * @param {String} key - translation key
+   * @param {*=} params
+   * @returns {*|string}
+   */
+  // eslint-disable-next-line no-unused-vars
+  i18n (key, params = {}) {
```

Comment:
might be a good idea to extract this

Reply:
can't see any other place since it needs `this.req.language`